eras/allegra/impl/cardano-ledger-allegra.cabal

```diff
@@ -24,6 +24,45 @@ source-repository head
     subdir:   eras/allegra/impl
 
 library
+    reexported-modules:
+        Cardano.Ledger.Allegra,
+        Cardano.Ledger.Allegra.Core,
+        Cardano.Ledger.Allegra.Rules,
+        Cardano.Ledger.Allegra.Scripts,
+        Cardano.Ledger.Allegra.Transition,
+        Cardano.Ledger.Allegra.Translation,
+        Cardano.Ledger.Allegra.Tx,
+        Cardano.Ledger.Allegra.TxAuxData,
+        Cardano.Ledger.Allegra.TxBody
+
+    default-language: Haskell2010
+    ghc-options:
+        -Wall -Wcompat -Wincomplete-record-updates
+        -Wincomplete-uni-patterns -Wredundant-constraints -Wpartial-fields
+        -Wunused-packages
+
+    build-depends:
+        base >=4.14 && <4.19,
+        aeson,
+        bytestring,
+        cardano-crypto-class,
+        cardano-ledger-binary >=1.0,
+        cardano-ledger-core >=1.9 && <1.11,
+        cardano-ledger-shelley >=1.7 && <1.10,
+        cardano-strict-containers,
+        cardano-slotting,
+        cborg,
+        containers,
+        deepseq,
+        internal,
+        microlens,
+        nothunks,
+        small-steps >=1.0,
+        transformers,
+        validation-selective
+
+library internal
+    visibility: private
```

Comment:
Why are you defining a whole new internal library?

And why are all the modules and dependencies listed twice?

We already have a `testlib` defined in all packages specifically for testing, so it would be enough to add `TestCardano.Ledger.Allegra.TreeDiff` module there

Reply:
Solved!